File: apps/backend/internal/announcement/model/announcement.go
package model

import "time"

const (
	TypeNormal      = "normal"
	TypeUpdate      = "update"
	TypeMaintenance = "maintenance"
	TypeWarning     = "warning"
	TypeEvent       = "event"
	TypeSystem      = "system"

	DisplayModeNormal       = "normal"
	DisplayModeBanner       = "banner"
	DisplayModeModal        = "modal"
	DisplayModeStartupModal = "startup_modal"

	TargetAll     = "all"
	TargetWeb     = "web"
	TargetAndroid = "android"
	TargetIOS     = "ios"
	TargetDesktop = "desktop"
)

type Announcement struct {
	ID          uint   `gorm:"primaryKey"`
	Title       string `gorm:"size:255;not null"`
	Content     string `gorm:"not null"`
	Type        string `gorm:"size:32;not null"`
	DisplayMode string `gorm:"size:32;not null"`
	Target      string `gorm:"size:32;not null"`
	Priority    int    `gorm:"not null"`
	StartsAt    *time.Time
	EndsAt      *time.Time
	Dismissible bool `gorm:"not null"`
	Published   bool `gorm:"not null"`
	CreatedBy   *uint
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

File: apps/frontend/app/types/announcement.ts
export type AnnouncementType =
  | 'normal'
  | 'update'
  | 'maintenance'
  | 'warning'
  | 'event'
  | 'system'

export type AnnouncementDisplayMode =
  | 'normal'
  | 'banner'
  | 'modal'
  | 'startup_modal'

export type AnnouncementTarget = 'all' | 'web' | 'android' | 'ios' | 'desktop'

export interface Announcement {
  id: number
  title: string
  content: string
  type: AnnouncementType
  displayMode: AnnouncementDisplayMode
  target: AnnouncementTarget
  priority: number
  dismissible: boolean
  published?: boolean
  startsAt?: string | null
  endsAt?: string | null
  createdBy?: number | null
  createdAt?: string
  updatedAt?: string
}

export interface AnnouncementPayload {
  title: string
  content: string
  type: AnnouncementType
  displayMode: AnnouncementDisplayMode
  target: AnnouncementTarget
  priority: number
  dismissible: boolean
  published: boolean
  startsAt?: string | null
  endsAt?: string | null
}
